### PR TITLE
[node] Add typings for stream.isReadable and stream.isErrored

### PR DESCRIPTION
**Pull request by sluukkonen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [17.x](https://nodejs.org/api/stream.html#streamisreadablestream) [16.x](https://nodejs.org/docs/latest-v16.x/api/stream.html#streamisreadablestream)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by sluukkonen** [comment]
As far as I can tell, the test failures aren't related to this PR.

**Comment by peterblazejewicz** [comment]
FYI: I've updated from master to fix CI failures with '@types/node' that landed master

File: types/node/stream.d.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ declare module 'stream' {
              * @since v16.8.0
              */
             static isDisturbed(stream: Readable | NodeJS.ReadableStream): boolean;
+            /**
+             * Returns whether the stream is readable.
+             * @since v17.4.0
+             */
+            static isReadable(stream: unknown): stream is Readable;
```
Comment (dnalborczyk) [inline types/node/stream.d.ts:64]: according to the docs: https://nodejs.org/api/stream.html#streamisreadablestream some type of `stream` is expected as parameter.

although passing anything but a stream returns `null`

```ts
require('stream').isReadable('foo') // null
```

interestingly, only passing a `ReadableStream` returns `true`, passing a `WriteableStream` returns also `null`, which might be a bug.

Comment (sluukkonen) [reply types/node/stream.d.ts:64]: Looks like I've misunderstood the intended functionality. Looking at the code in the [original PR](https://github.com/nodejs/node/pull/41199/files), it's designed to return whether a readable stream is in a readable state, not whether the stream is a readable stream.

Oh well, this makes the function not so useful for my intended use case (even though by chance it works anyway 😄 ).

I'll try to update the PR in the coming days.

Comment (dnalborczyk) [reply types/node/stream.d.ts:64]: I'm thinking the function should probably throw an error if anything else but a ReadableStream is passed in.

**Comment by sluukkonen** [comment]
PR has now been updated. I added typings for `stream.isErrored` while at it, since I noticed they were missing as well.

**Comment by gabritto** [comment]
@dnalborczyk @peterblazejewicz could you review the updates?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by sluukkonen** [comment]
Ready to merge

**Pull request merged**